baseapp/abci.go

```diff
@@ -901,6 +937,9 @@ func (app *BaseApp) checkHalt(height int64, time time.Time) error {
 // against that height and gracefully halt if it matches the latest committed
 // height.
 func (app *BaseApp) Commit() (*abci.ResponseCommit, error) {
+	app.mtx.Lock()
+	defer app.mtx.Unlock()
+
 	header := app.finalizeBlockState.ctx.BlockHeader()
 	retainHeight := app.GetBlockRetentionHeight(header.Height)
 
```

Comment:
Question: do we still need the `commiter` logic from [here](https://github.com/dydxprotocol/cosmos-sdk/blame/bdf96fddf421e90574e53af383c1468fdb33d7a3/baseapp/abci.go#L512-L514)?